### PR TITLE
Feature/dissco exporter

### PR DESCRIPTION
**Pull request by southeo**
Adds deployment files for export-backend and export-job
- Export-jobs run in their own namespace, so they need their own Service account (that needs to be set up in TF after these files are deployed)

### CODE REVIEW

File: test/data-export-job/job-request.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+kind: ConfigMap
```
Comment (samleeflang) [inline test/data-export-job/job-request.yaml:2]: What does the configmap do?

File: test/data-exporter-backend/data-exporter-backend-route.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    - match: Host(`dev.dissco.tech`) && PathPrefix(`/api`)
+      kind: Rule
+      services:
+        - name: data-exporter-backend-service
+          namespace: default
+          port: 8080
+      middlewares:
+        - name: data-exporter-backend-cors-headers
+---
+apiVersion: traefik.io/v1alpha1
```
Comment (samleeflang) [inline test/data-exporter-backend/data-exporter-backend-route.yaml:30]: Wait with the middleware as it doesn't work.
See the apiVersion `traefik.io/v1alpha1` for the middleware vs `traefik.containo.us/v1alpha1` for the rest.
Let's first upgrade to Traefik 3.X and then re-add the middleware (if necessary)

File: test/data-exporter-backend/data-exporter-backend.yaml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  replicas: 1
+  selector:
+    matchLabels:
+      app: data-exporter-backend
+  template:
+    metadata:
+      labels:
+        app: data-exporter-backend
+        language: java
+      annotations:
```
Comment (samleeflang) [inline test/data-exporter-backend/data-exporter-backend.yaml:18]: You can remove these as they were for logging to elastic and we will change this to logging to openTelemetry

File: test/data-exporter-backend/data-exporter-backend.yaml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        co.elastic.logs/multiline.pattern: '^[[:space:]]+(at|\.{3})[[:space:]]+\b|^Caused by:'
+        co.elastic.logs/multiline.negate: "false"
+        co.elastic.logs/multiline.match: after
+    spec:
+      serviceAccountName: secret-manager
+      automountServiceAccountToken: true
+      containers:
+        - name: data-exporter-backend
+          image: public.ecr.aws/dissco/dissco-core-backend
```
Comment (samleeflang) [inline test/data-exporter-backend/data-exporter-backend.yaml:27]: Just to make it clear add `:latest` to the image

File: test/data-export-job/aws-secrets.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+          key: col-bucket-access-key-secret
+  parameters:
+    objects: |
+      - objectName: "arn:aws:secretsmanager:eu-west-2:824841205322:secret:dissco-test-secrets-CkrWsm"
+        jmesPath:
+            - path: "elastic_username"
+              objectAlias: "elastic-username"
+            - path: "elastic_password"
+              objectAlias: "elastic-password"
+            - path: "mongodb_connection"
```
Comment (samleeflang) [inline test/data-export-job/aws-secrets.yaml:28]: We don't need mongo here right? just elastic and the bucket access key or something

**Review by samleeflang — COMMENTED** [review]
Couple things to check
Oh and for the secrets don't forget we need another entry in the terraform files

File: test/data-export-job/job-request.yaml (hunk 1 above)
Comment (southeo) [reply test/data-export-job/job-request.yaml:2]: sorry, this should be deleted

File: test/data-exporter-backend/data-exporter-backend-route.yaml [hunk 6]
```diff
@@ -25,19 +25,4 @@ spec:
...
-    accessControlAllowMethods:
-      - "GET"
-    accessControlAllowHeaders:
-      - "*"
-    accessControlAllowOriginList:
-      - "*"
-    accessControlMaxAge: 100
-    addVaryHeader: true
\ No newline at end of file
+        - name: data-exporter-backend-cors-headers
```
Comment (samleeflang) [inline test/data-exporter-backend/data-exporter-backend-route.yaml:28]: This looks like it still will look for middlewares

**Review by samleeflang — APPROVED** [review]
:rabbit: Just one point to check before merging

**Pull request merged**